### ISSUE TITLE
Use MAIN_MODULE=2 can not export function like '_Znwm'

### ISSUE THREAD

**Issue by deepsadness** [emscripten-core/emscripten#22028]
Please include the following in your bug report:

**Version of emscripten/emsdk:**
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.55 (165133b1cc977f0b9a277e42ef809b823157189c)
clang version 19.0.0git (https:/github.com/llvm/llvm-project 6c7805d5d186a6d1263f90b8033ad85e2d2633d7)
Target: wasm32-unknown-emscripten
Thread model: posix

**Failing command line in full:**
```
emcc cat.c -I . --pre-js pre.js -s EXPORTED_RUNTIME_METHODS='['ccall','cwrap','FS']' -s EXPORTED_FUNCTIONS='['_crc32','_Znwm']' -s USE_ZLIB=1 -s FORCE_FILESYSTEM=1 -s USE_PTHREADS=1 -pthread -s MAIN_MODULE=2 -s ERROR_ON_UNDEFINED_SYMBOLS=0 -s INITIAL_MEMORY=33554432 -s EXPORT_ES6=1 -s FULL_ES3=1 -s MODULARIZE=1 -s EXPORT_NAME=cat -s VERBOSE=1
```
result:
<img width="751" alt="image" src="https://github.com/emscripten-core/emscripten/assets/15323984/68e409f0-5684-4b26-b929-ab0d89dcf1a6">

**Comment by sbc100** [comment]
`_Znwm` is the C symbol name.  You need to add another underscore to the name if you want to include it in `EXPORTE_FUNCTIONS`

**Comment by deepsadness** [comment]
Thank a lot. It really helps!

But, How can i export `__cxa_end_catch`? 

EXPORTED_FUNCTIONS
```
EXPORTED_FUNCTIONS="['_crc32','___THREW__','__Znwm','___cxa_end_catch']"
```

**result:**
<img width="657" alt="image" src="https://github.com/emscripten-core/emscripten/assets/15323984/736d4e76-e3aa-4181-a1c1-db1b115e8052">

**Comment by sbc100** [comment]
Exporting `__cxa_end_catch` seems to work fine for me:

```
$ ./emcc test/hello_world.c -sEXPORTED_FUNCTIONS=__Znwm,_main,___cxa_end_catch
```

The above command seems to work for me both with and without `-fwasm-exceptions`.

Can you share the full set of link flags you are using?

BTW, for settings like `EXPORTED_FUNCTIONS` you don't need the surrounding quotes or square brackets anymore.

**Comment by deepsadness** [comment]
Thanks!
it work for me with `-fwasm-exceptions`.
```
emcc cat_cc.cc -sEXPORTED_FUNCTIONS=__Znwm,___cxa_end_catch -fwasm-exceptions
```
result: no error no warning!


but without `-fwasm-exceptions`. it still reports error.
```
emcc cat_cc.cc -sEXPORTED_FUNCTIONS=__Znwm,___cxa_end_catch
```
result:
<img width="692" alt="image" src="https://github.com/emscripten-core/emscripten/assets/15323984/e44bc0ab-1ad5-44ba-9673-ea64b595f059">

**Comment by sbc100** [comment]
What versions of emscripten are you using ?

**Comment by sbc100** [comment]
(Oh wait I see 3.1.55 in the OP.. I'll see if I can repro)

**Comment by sbc100** [comment]
Ah I see the problem, the issue is that this is JS symbol (without `-fwasm-exceptions`)  and back in 3.1.55 if you want to export a JS library symbol you would also need to add it to `-sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE=__cxa_end_catch` (note that you don't need the extra leading underscore here).

This was fixed in #21867 which was released in 3.1.60, so once you upgrade to 3.1.60 you won't need `DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` anymore.

**Comment by deepsadness** [comment]
Thanks a lot !! 
<img width="52" alt="image" src="https://github.com/emscripten-core/emscripten/assets/15323984/61b9132c-47fe-42d8-9744-7c4e23247ab7">
